### PR TITLE
Refactor listeners

### PR DESCRIPTION
**Pull request by ltatarev**
Feature refactors main 'module' to `useSharingIntent` hook which properly adds and removes listeners to relevant events.
iOS issues were caused by improper handling of Linking listener.

All the files were linted. Let me know if there are any more improvements to be added here.

I'll open an additional PR which contains changes got after building the library.

### CODE REVIEW

File: src/mimeTypes.ts
Comment (professorkolik) [inline src/mimeTypes.ts:1]: what was changed here?

File: src/useSharingIntent.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        ReceiveSharingIntent.getFileNames(url)
+          .then((data: any) => {
+            const files = sortData(data);
+            handler(files);
+          })
+          .catch((e: any) => errorHandler(e));
+      } else {
+        ReceiveSharingIntent.getFileNames()
+          .then((fileObject: any) => {
+            const files = Object.keys(fileObject).map((k) => fileObject[k]);
```
Comment (professorkolik) [inline src/useSharingIntent.ts:26]: You can move the content of this and the one in iOS into single function in which you can add a condition isIos

**Review by professorkolik — APPROVED** [review]

File: src/mimeTypes.ts
Comment (ltatarev) [reply src/mimeTypes.ts:1]: doublequotes to singlequotes :D I can revert the change if needed

File: src/useSharingIntent.ts (hunk 1 above)
Comment (ltatarev) [reply src/useSharingIntent.ts:26]: I tried adding it like that at first but I got some TS errors, will try again to see if I made some mistakes

**Review by swizes — APPROVED** [review]

File: src/mimeTypes.ts
Comment (professorkolik) [reply src/mimeTypes.ts:1]: no need
do you think it make sense to get from some package?

Comment (ltatarev) [reply src/mimeTypes.ts:1]: Yes, definitely. I'll check if the mime types package we use in the app already exports these consts

**Pull request merged**